CMakeLists.txt

```diff
@@ -6,73 +6,116 @@
 
   You should have received a copy of the license along with this work.  If not,
   see <http://creativecommons.org/licenses/by-sa/4.0/>.
+
+  Envoking CMake from the command line example usage:
+  cmake .. "-GUnix Makefiles" -DSYCL_ACADEMY_USE_DPCPP=ON -DSYCL_ACADEMY_BUILD_EXERCISES=2 
+  -DSYCL_ACADEMY_ENABLE_SOLUTIONS=ON -DCMAKE_C_COMPILER=icx -DCMAKE_CXX_COMPILER=icpx
 ]]
 
-cmake_minimum_required (VERSION 3.14)
+# Almost all CMake files should start with this
+# You should always specify a range with the newest
+# and oldest tested versions of CMake. This will ensure
+# you pick up the best policies. Minimum version required
+# by /opt/intel/oneapi/compiler/latest/linux/IntelSYCL/ReadNeSYCL.txt
+if(WIN32)
+  cmake_minimum_required(VERSION 3.23...3.27)
+else()
+  cmake_minimum_required(VERSION 3.20...3.27)
+endif()
+
```

Comment:
Following CMake modern good guidelines to layout and commenting. I learnt from it and as to its importance - it is not common knowledge.  The Intel advice has been taken. What would you advise in its place? My CMake version is 3.22.1.

Reply:
Hmmm, the file they provide does _say_ it requires those versions, I wonder why they don't actually specify that as a `cmake_minimum_required` line in the module :thinking:

I also checked the release notes and I am struggling to see what 3.23 brings for them. The reason I'm checking is because Ubuntu 20.04 (which is still heavily used) has CMake 3.16 by default. 22.04 ships with 3.22 (which obviously wouldn't be an issue for Windows) but again, I'd like not to diverge based on platform as much as is possible.

As for the comment, it's just a bit untidy when the code was already doing the right thing. I get the desire for explaining why things are a certain way but I think it's pretty well-understood by now.

Reply:
CMake min max requirement statement removed.

Reply:
CMake for Ubuntu does not go any higher than 3.22.1. If it did then the check for which platform could be removed Going lower may cause problems with Windows, so leave the split check in place.